src/lib/transformer.js

```diff
@@ -0,0 +1,197 @@
+import defaultConfig from '../config';
+import conditions from './conditions';
+
+function removeSVGTagAttrs(tag) {
+  if (conditions.isSVGToken(tag)) {
+    tag.attributes = tag.attributes.filter(conditions.hasNoWidthHeight);
+  }
+  return tag;
+}
+
+function createRemoveTagAttrs(removingTagAttrs = []) {
+  const hasNoAttributes = conditions.createHasNoAttributes(removingTagAttrs);
+  return function removeTagAttrs(tag) {
+    if (conditions.isStartTag(tag)) {
+      tag.attributes = tag.attributes.filter(hasNoAttributes);
+    }
+    return tag;
+  };
+}
+function createWarnTagAttrs(warnTagAttrs = []) {
+  const hasNoAttributes = conditions.createHasAttributes(warnTagAttrs);
+  return function warnTagAttrs(tag) {
+    if (conditions.isStartTag(tag)) {
+      const attrs = tag.attributes.filter(hasNoAttributes);
+      if (attrs.length > 0) {
+        const attrList = [];
+
+        for (const attr of attrs) {
+          attrList.push(attr[0]);
+        }
+
+        console.warn(
+          `svg-inline-loader: tag ${
+            tag.tagName
+          } has forbidden attrs: ${attrList.join(', ')}`
+        );
+      }
+    }
+    return tag;
+  };
+}
+function isRemovingTag(removingTags, tag) {
+  return removingTags.includes(tag.tagName);
+}
+function isWarningTag(warningTags, tag) {
+  return warningTags.includes(tag.tagName);
+}
+// FIXME: Due to limtation of parser, we need to implement our
```

Comment:
This loader is also a candidate for `posthtml` and the current logic implemented as `posthtml-plugins` instead, once I figured out what's going on here in particular :). The current code is just crazy...